Add OpInfo for `nn.functional.poisson_nll_loss`

ejguan opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #67465 Add OpInfo for `nn.functional.cosine_embedding_loss`
* #67464 Add OpInfor for `nn.functional.ctc_loss`
* **#67371 Add OpInfo for `nn.functional.poisson_nll_loss`**
* #67376 Add OpInfo for `nn.functional.huber_loss`
* #67381 Add OpInfo for `nn.functional.hinge_embedding_loss`
* #67356 Add OpInfo for `torch.nn.functional.gaussian_nll_loss`

Differential Revision: [D31973173](https://our.internmc.facebook.com/intern/diff/D31973173)

ejguan committed:
Add OpInfo for `nn.functional.poisson_nll_loss`

[ghstack-poisoned]

ejguan committed:
Update on "Add OpInfo for `nn.functional.poisson_nll_loss`"


Differential Revision: [D31973173](https://our.internmc.facebook.com/intern/diff/D31973173)

[ghstack-poisoned]

mruberry reviewed (approved):
Filing and linking an issue for the failing tests would be great. 

Otherwise LGTM! One inline comment about possibly adding a sample input that uses eps.

ejguan committed:
Update on "Add OpInfo for `nn.functional.poisson_nll_loss`"


Differential Revision: [D31973173](https://our.internmc.facebook.com/intern/diff/D31973173)

[ghstack-poisoned]

ejguan committed:
Update on "Add OpInfo for `nn.functional.poisson_nll_loss`"


Differential Revision: [D31973173](https://our.internmc.facebook.com/intern/diff/D31973173)

[ghstack-poisoned]
